lib/IRGen/GenCall.cpp

```diff
@@ -2605,9 +2606,16 @@ class SyncCallEmission final : public CallEmission {
     if (fnConv.getNumDirectSILResults() == 1
         && (fnConv.getDirectSILResults().begin()->getConvention()
             == ResultConvention::Autoreleased)) {
-      if (IGF.IGM.Context.LangOpts.EnableObjCInterop)
-        result = emitObjCRetainAutoreleasedReturnValue(IGF, result);
-      else
+      if (IGF.IGM.Context.LangOpts.EnableObjCInterop) {
```

Comment:
Why do we need the EnableObjCInterop check here? The fact that custom reference counting currently only kicks in with C++ (is that even true?) might not be correct in the future, and we should be able to rely on the `ReferenceCounting::Custom` check to correctly identify the cases.